packages/wrangler/src/vectorize/insert.ts

```diff
@@ -30,13 +41,32 @@ export async function handler(
 	args: StrictYargsOptionsToInterface<typeof options>
 ) {
 	const config = readConfig(args.config, args);
+	const file = await open(args.file);
 
-	const vectors: VectorizeVector[] = [];
-	if (args.vectors) {
-		// Parse each vector into a Vector type
-		// Think about potential limits on args.vectors.length?
+	let index: Optional<VectorizeVectorMutation, "ids"> | undefined;
+	for await (const batch of getBatchFromFile(file, args.batchSize)) {
+		const formData = new FormData();
+		formData.append("vectors", new File([batch.join(`\n`)], "vectors.ndjson"));
+		logger.log(`✨ Uploading vector batch (${batch.length} vectors)`);
+		const idxPart = await insertIntoIndex(config, args.name, formData);
+		if (!index) index = idxPart;
+		else index.count += idxPart.count;
 	}
 
-	const index = await insertIntoIndex(config, args.name, vectors);
+	// remove the ids - skip tracking these for bulk uploads since this could be in the 100s of thousands.
```

Comment:
I think this is probably the right move for this type of operation, but feel free to shout if you feel differently